tests/functional/fetchr.test.js

```diff
@@ -266,6 +266,25 @@ describe('client/server integration', () => {
                 });
             });
 
+            it('can abort after a timeout', async () => {
+                const response = await page.evaluate(() => {
+                    const fetcher = new Fetchr({});
+                    const promise = fetcher.read('slow', null, {
+                        retry: { maxRetries: 5, interval: 0 },
+                        timeout: 200,
+                    });
+
+                    return new Promise((resolve) =>
+                        setTimeout(() => {
+                            promise.abort();
+                            resolve();
+                        }, 500)
+                    ).then(() => promise.catch((err) => err));
+                });
+
+                expect(response.reason).to.equal(FetchrError.ABORT);
+            });
```

Comment:
@redonkulus this was the test that was failing with the previous implementation. It's ugly, but, basically, it triggers a call to the slow resource (which always takes 5s to respond) with a timeout of 200ms. After that, we schedule an `abort` call after 500ms (in the middle of the 3rd call).

This test would fail since we would get a `TIMEOUT` error (meaning that we were aborting a controller that was already stale). But now, the correct error is sent (meaning that we were able to prevent the last 2 calls to happen).

Reply:
again, i like more details in the code than not, feel free to add this scenario as a comment too